devices/IFrameTransformToIWear/src/IFrameTransformToIWear.cpp

```diff
@@ -0,0 +1,518 @@
+/*
+ * Copyright (C) 2021 Istituto Italiano di Tecnologia (IIT)
+ * All rights reserved.
+ *
+ * This software may be modified and distributed under the terms of the
+ * GNU Lesser General Public License v2.1 or any later version.
+ */
+
+#include "IFrameTransformToIWear.h"
+
+#include <yarp/dev/IFrameTransform.h>
+#include <yarp/os/LogStream.h>
+#include <yarp/sig/Vector.h>
+
+#include <map>
+#include <mutex>
+#include <string>
+#include <vector>
+
+const std::string DeviceName = "IFrameTransformToIWear";
+const std::string LogPrefix = DeviceName + " :";
+
+using namespace wearable;
+using namespace wearable::sensor;
+using namespace wearable::devices;
+
+struct ParsedOptions
+{
+    wearable::WearableName wearableName;
+    wearable::sensor::SensorType wearableSensorType;
+    std::string rootFrameID;
+
+    std::vector<std::string> frameIDs;
+};
+
+// Class that provides each frame information, and utilites
+// to map its data to containers compatible with IWear
+class IFrameTransformHandler
```

Comment:
I don't see the initialization of the yarp network. If this is necessary, you also to link against `YARP::YARP_init`.